module/VuFind/src/VuFind/ServiceManager/Factory/Autowire.php

```diff
@@ -47,17 +47,20 @@ class Autowire
     /**
      * Constructor.
      *
-     * Use only one of the following parameters.
-     *
-     * @param string $config  Configuration to inject (as an array)
-     * @param string $service Service to inject
+     * @param string $config    Configuration to inject as an array (mutually exclusive with $service)
+     * @param string $service   Service to inject (mutually exclusive with $config)
+     * @param string $container Container or plugin manager to use to get the service
      */
     public function __construct(
         public readonly ?string $config = null,
         public readonly ?string $service = null,
+        public readonly ?string $container = null,
     ) {
         if (null !== $config && null !== $service) {
             throw new LogicException('#[Autowire] attribute cannot contain both config and service.');
         }
+        if (null !== $container && null === $service) {
+            throw new LogicException('#[Autowire] attribute cannot contain container without a service.');
```

Comment:
Is this restriction strictly necessary? If the argument is typed for a plugin found in the container, do we need to repeat that type as a service in the attribute, or could we just pull it from the type declaration? Maybe it's better to fail in the factory if we get a container and can't figure out a service name.